server/src/main/java/com/cloud/server/ManagementServerImpl.java

```diff
@@ -5493,4 +5500,24 @@ public void setLockControllerListener(final LockControllerListener lockControlle
         _lockControllerListener = lockControllerListener;
     }
 
+    @Override
+    @DB
+    @ActionEvent(eventType = EventTypes.EVENT_MANAGEMENT_SERVER_REMOVE, eventDescription = "removing Management Server")
+    public boolean removeManagementServer(RemoveMgmtCmd cmd) {
+        final Long id = cmd.getId();
+        ManagementServerJoinVO managementServer = _managementServerJoinDao.findById(id);
+
+        if (managementServer == null) {
+            throw new InvalidParameterValueException(String.format("Unable to find a Management Server with ID equal to [%s]", managementServer.getUuid()));
+        }
+
+        if (ManagementServerHost.State.Up.equals(managementServer.getState())) {
```

Comment:
```suggestion
        if (!ManagementServerHost.State.Down.equals(managementServer.getState())) {
```
If it should only be removed in the Down state ( as said in the logging message below ) this condition doesn't cover all cases, as we have more states than Up and Down.